### PR TITLE
less output, allow VERBOSE mode

### PR DESCRIPTION
**Pull request by planthaber**
As requested here: https://github.com/dfki-ric/docker_image_development/issues/16

Adds a "VERBOSE=true" setting that suppresses most of the exec scritpt output.

The environment variable "VERBOSE" can be set via settings.bash or directly in the command line:
```bash
VERBOSE=true ./exec.bash
```

### CODE REVIEW

**Comment by planthaber** [comment]
I will still give some output every time a new container is created:

```bash

Setting the containers devel user id to host user id
UID did not need to be changed

please run /opt/setup_workspace.bash to initialize the workspace
```

**Comment by 0Nel** [comment]
Great solution!!

For me I am still getting `WARNING: No swap limit support`. We could simply append `2> /dev/null`to the `init_docker $@` call in exec.bash, but that would suppress all error messages.

Considering this I would propose to have different levels of verbosity? One that really suppresses everything but the output of the command that is executed in the container (e.g. SILENT). Another one that suppresses general docker output, but still shows stuff like `using developmentimage/plain_20.04:base` (e.g. DEFAULT) and finally one that prints everything as it were before (e.g. VERBOSE).

**Comment by 0Nel** [comment]
For the SILENT one it would of course be nice to also suppress output when a new container is generated..

**Comment by planthaber** [comment]
> I would propose to have different levels of verbosity

I wanted to explain it here in detail, but it was faster to implement it :-)

To silence/adjust the output from the image, we'd have to provide the print variables in the docker run command, thus fixing those as long the container is not re-generated. (Currently I only suppress the output of the "outside docker scripts")

```bash
UID did not need to be changed

devel user id set, logging out automatically


please run /opt/setup_workspace.bash to initialize the workspace
```
Comes from the base images

File: settings.bash [hunk 1]
```diff
@@ -48,3 +48,10 @@ export WORKSPACE_RELEASE_IMAGE=developmentimage/${PROJECT_NAME}:release
 # --privileged
 # -v /dev/input/:/dev/input
 export ADDITIONAL_DOCKER_RUN_ARGS=""
+
+# Make the exec script to talk more for debugging/docker setup purposes.
+# This may also be stated in the command line: $> VERBOSE=true ./exec.bash 
+# export VERBOSE=true
```
Comment (0Nel) [inline settings.bash:54]: We should have either VERBOSE or SILENT uncommented right?

Apart from that I'd be okay to merge.

**Review by 0Nel — CHANGES_REQUESTED** [review]

File: settings.bash (hunk 1 above)
Comment (planthaber) [reply settings.bash:54]: No, in nothing is uncommented, it is the default (less output, only the image used and possible re-creations of the container).
There acually is a error message if both are set

Comment (0Nel) [reply settings.bash:54]: Okay I see :+1:

**Comment by 0Nel** [comment]
Should we start updating the version numbers?

I think we passed some backwards incompatible points already, but does it make sense in your experience to track development?

**Comment by planthaber** [comment]
Currently the sccripts should be backward compatible with the images on hub.docker.com

Also there is a warning, if the most recent change was not pulled by the scripts ans restored the "old" bevavior:
https://github.com/dfki-ric/docker_image_development/blob/56bc2ce3c5ea891b340374d3897799166f8e9802/image_setup/01_base_images/entrypoint.bash#L9

But ultimately I think it is a good idea to 
1. Start versioning the scripts
2. "bake" the script version that generated the base image into the base image
3. Give the scripts version as a parameter to docker run
4. Check if the scripts version fits the image

But in the end these issues can only occur when: 
1. A new base image is pulled AND
2. A new devel image is generated
or
1. The scritps are updated without regenerating the devel image

Otherwise the non-base images should stay compatible with the scripts, as they would use the layers of the older base imge.

**Pull request merged**